vllm/sampling_params.py

```diff
@@ -158,6 +165,14 @@ def _verify_greedy_sampling(self) -> None:
         if self.top_k != -1:
             raise ValueError("top_k must be -1 when using greedy sampling.")
 
+    @property
+    def sampling_type(self) -> SamplingType:
+        if self.use_beam_search:
+            return SamplingType.BEAM
+        if self.temperature < _SAMPLING_EPS:
+            return SamplingType.GREEDY
+        return SamplingType.RANDOM
```

Comment:
I have noticed during profiling that constantly calling this property becomes somewhat expensive, I would recommend instead turning it into a simple attribute as the last thing in the `__init__`

Reply:
Fixed with `cached_property`